test_runner/fixtures/neon_fixtures.py

```diff
@@ -4809,6 +4813,33 @@ def wait_for_last_flush_lsn(
     return min(results)
 
 
+def wait_for_commit_lsn(
+    env: NeonEnv,
+    tenant: TenantId,
+    timeline: TimelineId,
+    lsn: Lsn,
+) -> Lsn:
+    # TODO: it would be better to poll this in the compute, but there's no API for it. See:
+    # https://github.com/neondatabase/neon/issues/9758
+    "Wait for the given LSN to be committed on any Safekeeper"
+
+    max_commit_lsn = Lsn(0)
+    for i in range(1000):
```

Comment:
nit: would be nice to express this as a `wait_until` for consistency with other places we wait for things

Reply:
`wait_until` outputs status with the same frequency as the check. Here, I need relatively high-frequency checks (100 ms) to measure the ingestion duration. However, outputting status every 100 ms is very spammy, considering this will run for 10+ seconds.

I'll submit a follow-up PR to improve `wait_until` and clean up some duplicate logic here.